js/src/setup-ads/ads-stepper/index.js

```diff
@@ -27,12 +28,18 @@ import {
  */
 const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const { hasFinishedResolution, hasGoogleAdsConnection } =
```

Comment:
@kt-12 As per your comment here: https://github.com/woocommerce/google-listings-and-ads/issues/2534#issuecomment-2353038593, there's no page reload. Instead the `AdsStepper` component gets re rendered because `hasGoogleAdsConnection` has been modified once you connect a Google Ads account. Instead of using the `useGoogleAdsAccount` hook, which will cause the component to render again whenever there's a change, I _think_ we can use the `glaData.initialWpData.adsId` global variable which is static and set upon page load. So upon page load, if `glaData.initialWpData.adsId` is not `null` then it means that there's a connected Google Ads account. Even when you connect an Ads account, `glaData.initialWpData.adsId` will still be `null`. So you can have something like this:
```js
const hasGoogleAdsId = glaData.initialWpData.adsId;
```

Reply:
This was new to me, but it defiantly did the trick.